### PR TITLE
Added support for skip() and limit()

### PR DESCRIPTION
**Pull request by siddeshwarnavink**
Hay there,
I have added the skip() and limit() feature which I found was missing while I was working on a project with **pagination support**.

```js
// Skip
const skipTwo = await users.skip(2).find();

// Limit
const featuredUser = await users.limit(5).find();
```

Expecting a positive review 👍

### CODE REVIEW

File: ts/collection.ts [hunk 1]
```diff
@@ -377,4 +387,16 @@ export class Collection<T extends any> {
...
+  public limit(maxLimit: number) {
+    this.maxQueryLimit = maxLimit;
+    return this;
+  }
+
+  // Skip
+  public skip(skipCount: number) {
+    this.skipDocCount = skipCount;
+    return this;
+  }
```
Comment (manyuanrong) [inline ts/collection.ts:401]:
```ts
const users = db.collection<UserSchema>("users");
const docs1 = users.skip(2).limit(10).find();

// The execution of this line of code will be incorrectly affected.
const docs2 = users.find();
```

We can only put limit and skip into the second parameter of the find method. Or the find method returns a special promise after execution, which has limit and skip methods.

Comment (siddeshwarnavink) [reply ts/collection.ts:401]: If I got you correctly, you're expecting something like this right?
```js
const users = db.collection<UserSchema>("users");
const docs1 = users.find({}, { skip: 2, limit: 10 });
// Or
const docs2 = users.find().skip(2).limit(10);
```

Comment (manyuanrong) [reply ts/collection.ts:401]: Yes correct

**Comment by siddeshwarnavink** [comment]
Hay there,
I am getting a strange error while running the CI test on windows while merging my commit with the pull request.

```
test result: ok. 18 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out (309ms)

##[error]Process completed with exit code 1.
```

It is showing `` 0 failed`` but still, it completed with exit code 1 and fails the CI test. Please let me know if there is any problem with my code or with something else.

**Comment by manyuanrong** [comment]
> Hay there,
> I am getting a strange error while running the CI test on windows while merging my commit with the pull request.
> 
> ```
> test result: ok. 18 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out (309ms)
> 
> ##[error]Process completed with exit code 1.
> ```
> 
> It is showing ` 0 failed` but still, it completed with exit code 1 and fails the CI test. Please let me know if there is any problem with my code or with something else.

The problem has lasted for a long time, and the cause has not been found

File: ts/collection.ts [hunk 2]
```diff
@@ -196,11 +202,50 @@ export class Collection<T extends any> {
...
+        return this._get().then(callback);
+      }
+
+      public catch(callback: any): Promise<(T & WithID) | null> {
+        return this._get().catch(callback);
+      }
+
+      public finally(callback: any): Promise<(T & WithID) | null> {
+        return this._get().finally(callback);
+      }
```
Comment (manyuanrong) [inline ts/collection.ts:237]: If you execute 
```ts
users.find().skip(1).then(console.log).catch(console.error);
```

`then` and `catch` each executed a query, which is not expected which is not expected

File: README.md [hunk 3]
```diff
@@ -95,4 +95,20 @@ const deleteCount = await users.deleteOne({ _id: insertId });
...
+const featuredUser = await users.find().limit(5);
+
+// Skip and Limit
+const users = await users.find().skip(5).limit(5);
+=======
+const skipTwo = await users.skip(2).find();
+
+// Limit
+const featuredUser = await users.limit(5).find();
+>>>>>>> limit-and-skip
```
Comment (manyuanrong) [inline README.md:113]: Here should be cleaned up

Comment (siddeshwarnavink) [reply README.md:113]: Ooops. I didn't see that coming 😅

**Pull request merged**